core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -144,7 +144,7 @@ public static HighlightBuilder highlight() {
 
     private Float minScore;
 
-    private long timeoutInMillis = -1;
+    private TimeValue timeout = TimeValue.MINUS_ONE;
```

Comment:
That makes sense when the field is a `long` but when you change it to some other type it makes less sense. I guess I'm wondering if now is the right time to change that. At least here.